plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/packagejson/PackageJsonValidatorExtension.java

```diff
@@ -166,66 +181,100 @@ public void checkName(JSONValue projectNameValue) {
-				final IResource resolvedResource = root.findMember(platformURI);
-				if (resolvedResource instanceof IProject) {
-					platformProjectContainer = resolvedResource.getName();
-				}
-				// check project name to match file system folder name
-				final String fileSystemName = resolvedResource.getLocation().lastSegment();
-				if (!fileSystemName.equals(projectName.getValue())) {
-					addIssue(
-							IssueCodes.getMessageForPKGJ_PACKAGE_NAME_MISMATCH(projectName.getValue(), fileSystemName),
-							projectName, IssueCodes.PKGJ_PACKAGE_NAME_MISMATCH);
-				}
-			} else {
-				// otherwise, assume file-based URI
-				platformProjectContainer = new File(packageJsonUri.toFileString()).getParentFile().getName();
-			}
 
-			if (platformProjectContainer == null) {
-				// Container project cannot be determined, fail gracefully. We currently assume that this
-				// case will only occur when a nested package.json is validated in an editor (we ignore
-				// nested package.json files in the builder)
+			if (nameInfo.eclipseProjectName == null) {
+				// Eclipse project name cannot be determined, fail gracefully. We currently assume that this case will
+				// only occur when a nested package.json is validated in an editor (we ignore nested package.json files
+				// in the builder)
 				return;
 			}
 
-			// check project name to match Eclipse workspace project name
-			if (!platformProjectContainer.equals(projectName.getValue())) {
-				final String message = IssueCodes.getMessageForPKGJ_PROJECT_NAME_ECLIPSE_MISMATCH(
-						projectName.getValue(),
-						platformProjectContainer);
-				addIssue(message, projectName,
-						IssueCodes.PKGJ_PROJECT_NAME_ECLIPSE_MISMATCH);
-			}
-		} else {
-			// Make sure the package name equals the package folder name.
-			// Assumption: since the platform is not running, the URI actually represents the file system hierarchy
-			final String packageUriFolderName = projectNameValue.eResource().getURI().trimSegments(1).lastSegment();
-			if (!packageUriFolderName.equals(projectName.getValue())) {
-				addIssue(
-						IssueCodes.getMessageForPKGJ_PACKAGE_NAME_MISMATCH(projectName.getValue(),
-								packageUriFolderName),
-						projectName, IssueCodes.PKGJ_PACKAGE_NAME_MISMATCH);
+			String expectedEclipseProjectName = ProjectDescriptionUtils
+					.convertN4JSProjectNameToEclipseProjectName(projectName);
+			if (!expectedEclipseProjectName.equals(nameInfo.eclipseProjectName)) {
+				String msg = IssueCodes.getMessageForPKGJ_PROJECT_NAME_ECLIPSE_MISMATCH(
+						expectedEclipseProjectName, nameInfo.eclipseProjectName);
+				addIssue(msg, projectNameLiteral, IssueCodes.PKGJ_PROJECT_NAME_ECLIPSE_MISMATCH);
 			}
 		}
+	}
+
+	private static final boolean isValidNpmPackageName(String name) {
+		return !Strings.isNullOrEmpty(name)
+				&& name.length() <= 214
+				&& !NPM_RESERVED_PACKAGE_NAMES.contains(name)
+				&& IDENTIFIER_PATTERN.matcher(name).matches();
+	}
 
+	private static final class ProjectNameInfo {
```

Comment:
Maybe this class can be of use somewhere else, too? (and could be made a utility)

Reply:
done